### PR TITLE
feat(ci): automate deploy of npm packages to npm registry

### PR DESCRIPTION
**Pull request by vdovhanych**
This pr is supposed to automate publishing suite NPM packages to the NPM registry. 

Todo: 
- [x] add npm token
- [x] add jobs for releasing beta versions
- [ ] ci check script for preventing releasing bad versions or beta version to release. @szymonlesisz

### CODE REVIEW

File: ci/deploy.yml [hunk 1]
```diff
@@ -80,3 +80,46 @@ msg-system deploy dev:
     - deploy
   script:
     - aws s3 cp packages/suite-data/files/message-system/config.v1.jws s3://data.trezor.io/config/develop/config.v1.jws
+
+# npm packages deploy to npm registry
+link deploy npm:
```
Comment (tsusanka) [inline ci/deploy.yml]: Let's definitely rename this to `blockchain-link` as at first I thought you mean trezor-link.

File: ci/deploy.yml [hunk 2]
```diff
@@ -80,3 +80,46 @@ msg-system deploy dev:
...
+    <<: *run_everything_rules
+  when: manual
+  before_script:
+    - echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
+  script:
+    - nix-shell --run "yarn && cd ./packages/blockchain-link && npm publish"
+  after_script:
+    - rm ~/.npmrc
+  tags:
+    - deploy
```
Comment (tsusanka) [inline ci/deploy.yml]: Let's move all the common things to an anchor:

```
  when: manual
  before_script:
    - echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
  after_script:
    - rm ~/.npmrc
  tags:
    - deploy
```

**Review by tsusanka — COMMENTED** [review]
@szymonlesisz btw why do we need blockchain-link and components in NPM? I guess blockchain-link is being used in Connect same as rollout? And components?

File: ci/deploy.yml (hunk 2 above)
Comment (tsusanka) [reply ci/deploy.yml]: @vdovhanych why not `tags` as well?

Comment (vdovhanych) [reply ci/deploy.yml]: I like when i can see it right away on what type of the runner it's running if you search for something. But as i think everything in the deploy.yml file should run on deploy tags so we could add this as an anchor for everything there.

Comment (tsusanka) [reply ci/deploy.yml]: You can use this argument for basically anything :). Other people might like to see `after_script`. I would be consistent in this.

Comment (vdovhanych) [reply ci/deploy.yml]: So i should keep the tag there as we want to have it consistent with the rest of the files?

Comment (tsusanka) [reply ci/deploy.yml]: Sorry, I meant consistency with the anchor :laughing:. Let's just add it to the anchor and I'll shut up.

Comment (vdovhanych) [reply ci/deploy.yml]: Yep added it to there

Comment (tsusanka) [reply ci/deploy.yml]: Eeegh I still do not see the change? :thinking:

Comment (vdovhanych) [reply ci/deploy.yml]: 🙈  should be fixed now

File: ci/deploy.yml [hunk 3]
```diff
@@ -80,3 +80,37 @@ msg-system deploy dev:
...
+  before_script:
+    - echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
+  after_script:
+    - rm ~/.npmrc
+
+blockchain-link deploy npm:
+  stage: deploy to dev
+  only:
+    <<: *run_everything_rules
```
Comment (tsusanka) [inline ci/deploy.yml]: I have just realized there is no reason not to include these three lines into the anchor as well. Sorry, last thing! Feel free to force-push then.

**Comment by vdovhanych** [comment]
Force pushed the change and also added the beta job, didn't put the run everything rules anchor inside anchor so we can use it for beta jobs as well which are not supposed to run from develop only.

File: ci/deploy.yml (hunk 3 above)
Comment (vdovhanych) [reply ci/deploy.yml]: Didn't put the anchor there as we want to use it for beta jobs as well and it would prevent them from running on any branch.

**Review by tsusanka — APPROVED** [review]

**Pull request merged**